### PR TITLE
Module 2 Exercise

### PR DESCRIPTION
**Pull request by pcoulso1**
I had some issues with using the date time picker and would like to see how it should be done.

Do you have an example ?

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,36 +1,48 @@
 from flask import Flask, render_template, request, redirect, url_for
-import session_items as session
+import trello_items as item_store
+
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (Jonesey13) [inline app.py]: I don't think you need this line as `flask run` will load the `.env` file automatically for you

File: app.py [hunk 2]
```diff
@@ -1,36 +1,48 @@
...
 
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
```
Comment (Jonesey13) [inline app.py]: You can remove this line and `flask_config.py` (as it's only used for the `SECRET_KEY` config value which is only used by the Flask session cookie)

File: trello_items.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
+import os
+import requests
+from item import Item
+
+"""
+A Global to be used to cache the doard ID
```
Comment (Jonesey13) [inline trello_items.py]: typo in `doard`

File: trello_items.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    Gets the board ID which is to be used for this application from Trello id not already cached
+
+    Returns:
+        id: The board ID.
+    """
+    global BOARD_ID
+    if BOARD_ID == "":
+        boards = get_boards()
+        BOARD_ID = next(
+            (board['id'] for board in boards if board['name'] == "ToDoBoard"), None)
```
Comment (Jonesey13) [inline trello_items.py]: Should the fallback be `""` here?

File: trello_items.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    return BOARD_ID
+
+
+def get_lists(include_cards=True):
+    """
+    Fetches all lists from the board in Trello.
+
+    Args:
+        include_cards: A flag to indicate is the items in the list should also be returned.
```
Comment (Jonesey13) [inline trello_items.py]: *if

File: trello_items.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    """
+    Fetches all saved items from the Trello.
+
+    Returns:
+        list: The list of saved items.
+    """
+    card_lists = get_lists()
+
+    items = []
+    for card_list in card_lists:
```
Comment (Jonesey13) [inline trello_items.py]: Probably easiest to flatten this out using a comprehension: https://stackoverflow.com/questions/952914/how-to-make-a-flat-list-out-of-list-of-lists

File: trello_items.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+    Returns:
+        list: The list of saved items.
+    """
+    board_id = get_board_id()
+
+    params = get_default_params()
+    if include_cards:
+        params.update({'cards': 'all'})
+    url = build_url('/boards/%s/lists' % board_id)
```
Comment (Jonesey13) [inline trello_items.py]: Format strings might be clearer: `f'/boards/{board_id}/lists'`

**Review by Jonesey13 — APPROVED** [review]
Looks all functionally sound to me!

Left a few comments here and there but feel free to merge this PR in when ready 👍

File: app.py (hunk 1 above)
Comment (pcoulso1) [reply app.py]: Removed

File: trello_items.py (hunk 3 above)
Comment (pcoulso1) [reply trello_items.py]: Fixed

File: app.py (hunk 2 above)
Comment (pcoulso1) [reply app.py]: Removed

File: trello_items.py (hunk 4 above)
Comment (pcoulso1) [reply trello_items.py]: Good point, I'll change this

File: trello_items.py (hunk 6 above)
Comment (pcoulso1) [reply trello_items.py]: I'll change

**Pull request merged**